lib/perl/Genome/VariantReporting/Command/Wrappers/GoldEvaluation.pm

```diff
@@ -86,5 +88,89 @@ sub is_single_bam {
     return (!defined($model->normal_sample));
 }
 
+sub get_common_translations {
+    my $self = shift;
+    my $model = $self->model;
+    if ($self->is_single_bam($model)) {
+        return $self->get_germline_translations();
+    }
+    else {
+        return $self->get_somatic_translations();
+    }
+}
+
+sub get_somatic_translations {
+    my $self = shift;
+
+    return {
+        sample_name_labels => {
+            $self->discovery_sample_name =>
+                sprintf('Discovery(%s)', $self->discovery_sample_name),
+            $self->normal_sample_name =>
+                sprintf('Normal(%s)', $self->normal_sample_name),
+            $self->gold_sample_name =>
+                sprintf('Gold(%s)', $self->gold_sample_name),
+        },
+        library_name_labels => {
+            $self->get_library_name_labels('discovery'),
+            $self->get_library_name_labels('normal'),
+        },
+    };
+}
+
+sub get_germline_translations {
+    my $self = shift;
+
+    return {
+        sample_name_labels => {
+            $self->discovery_sample_name =>
+                sprintf('Discovery(%s)', $self->discovery_sample_name),
+            $self->gold_sample_name =>
+                sprintf('Gold(%s)', $self->gold_sample_name),
+        },
+        library_name_labels => {
+            $self->get_library_name_labels('discovery'),
+        },
+    };
+}
+
+sub discovery_sample_name {
+    my $self = shift;
+    return $self->discovery_sample->name;
+}
+
+sub normal_sample_name {
+    my $self = shift;
+    return $self->normal_sample->name;
+}
+
+sub normal_sample {
+    my $self = shift;
+    return $self->model->normal_sample;
+}
+
+sub discovery_sample {
+    my $self = shift;
+    return $self->model->tumor_sample;
+}
+
+my %counters;
+sub get_library_name_labels {
```

Comment:
Perhaps.  I'm still not too familiar with this code base.  

Part of my "fixing" involved copying the logic from ModelReport and making "adjustments" from there.

Some of the nomenclature is a bit confusing to me.  For example, when we are supposed to be using the term "tumor" versus "discovery"?  To my initial eyes, the "discovery" sample should be really called the "tumor" sample.